src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -98,11 +102,13 @@ public boolean hasPerson(Person person) {
     @Override
     public void deletePerson(Person target) {
         addressBook.removePerson(target);
+        personList.remove(target);
     }
 
     @Override
     public void addPerson(Person person) {
         addressBook.addPerson(person);
+        personList.add(person);
```

Comment:
Hmm this might be a problem because the person is stored in 2 places - an ArrayList and UniquePersonList - increasing the logic. 

E.g, suppose the resident 1 has current student groups basketball. 
1. User adds choir using: `edit 1 s/basketball s/choir`
2. Then searches for all student groups via `list-group`

Will choir still show?

Perhaps you can consider removing this additional field, and then `return `addressbook.getPersonList()`? (Check `AddressBook.java` for this method).

Reply:
Right, it complicates the logic, I have updated it to `return addressbook.getPersonList()`. Thanks.